### PR TITLE
:seedling: Fixups for watch in in-memory apiServer

### PR DESCRIPTION
**Pull request by killianmuldoon**
Has a number of small improvements following on from reviews of #8851.

Related to #8814

### CODE REVIEW

File: test/infrastructure/inmemory/internal/server/api/handler.go [hunk 1]
```diff
@@ -73,29 +73,33 @@ func NewAPIServerHandler(manager cmanager.Manager, log logr.Logger, resolver Res
 
 	// CRUD endpoints (global objects)
 	ws.Route(ws.POST("/api/v1/{resource}").Consumes(runtime.ContentTypeProtobuf).To(apiServer.apiV1Create))
-	ws.Route(ws.GET("/api/v1/{resource}").To(apiServer.apiV1List))
+	ws.Route(ws.GET("/api/v1/{resource}").If(isList).To(apiServer.apiV1List))
```
Comment (killianmuldoon) [inline test/infrastructure/inmemory/internal/server/api/handler.go:76]: Route handling for Watch and List this way seems more natural and easy to follow this way IMO.

**Comment by killianmuldoon** [comment]
/retest

Flake caused by github not being reachable.

**Comment by sbueringer** [comment]
Thx!

/lgtm
/approve

/hold
Let's give others some time to review

**Pull request merged**